test/runtests.jl

```diff
@@ -1,6 +1,279 @@
+=#
+
+function genBAMFDD(T::Type{BAMF.ArrayDD}, sz::Int32=Int32(32))
+    return BAMF.ArrayDD(sz)
+end
+
+function genBAMFDD(T::Type{BAMF.DataSLIVER}, sz::Int32=Int32(32))
+    return BAMF.DataSLIVER(sz, [Int32(1)])
+end
+
+function genBAMFDD(T::Type{BAMF.AdaptData}, sz::Int32=Int32(32))
+    return BAMF.AdaptData(sz, [(BAMF.DDMeasType, ())])
+end
+
+#=
+DDinfo(DD::BAMF.BAMFData) returns a tuple of the information that ought to be contained in a direct
+detection BAMFData structure: size, an array of the appropriate dimensions, and the object itself.
+=#
+
+function DDinfo(DD::BAMF.BAMFData)
+    dim= size(DD.data)
+    sz= DD.sz
+    return sz, dim, DD
+end
+
+#=
+genBAMFSLIVER(T::Type{BAMF.BAMFData}, sz::Int32) returns an empty BAMFData SLIVER model of type T and
+size sz. The default size is 32.
+=#
+
+function genBAMFSLIVER(T::Type{BAMF.DataSLIVER}, sz::Int32=Int32(32))
+    return BAMF.DataSLIVER(sz, [Int32(2)])
+end
+
+function genBAMFSLIVER(T::Type{BAMF.AdaptData}, sz::Int32=Int32(32))
+    return BAMF.AdaptData(sz, [(BAMF.SLIVERMeasType, (Float32(0f0), Float32(0f0)))])
+end
+
+#=
+SLIVERinfo(SLIVER::BAMF.BAMFData) returns a tuple of the information that ought to be contained in a SLIVER
+BAMFData structure: size, an array of the appropriate dimensions, the object itself, and the x y inversion
+points alongside the integration time.
+=#
+
+function SLIVERinfo(SLIVER::BAMF.DataSLIVER)
+    dim= size(SLIVER.data)
+    sz= SLIVER.sz
+    inttime, invx, invy= SLIVER.inttime[1], SLIVER.invx[1], SLIVER.invy[1]
+    return sz, dim, inttime, invx, invy, SLIVER
+end
+
+function SLIVERinfo(SLIVER::BAMF.AdaptData)
+    dim= size(SLIVER.data)
+    sz= SLIVER.sz
+    inttime, invx, invy= SLIVER.meastypes[1].inttime, SLIVER.meastypes[1].invx, SLIVER.meastypes[1].invy
+    return sz, dim, inttime, invx, invy, SLIVER
+end
+
+#=
+genRJMCMC(burnin::Int32, iterations::Int32) is a function that produces an RJMCMCStruct with the usual DD accept 
+and propose functions alongside some standard jump probabilities. The default burnin and iterations is 1000.
+=#
+
+function genRJMCMC(burnin::Int32=Int32(1000), iterations::Int32=Int32(1000))
+    ## setup the RJMCMC.jl model
+    # Jumptypes are: move, bg, add, remove, split, merge
+    njumptypes=6
+    jumpprobability=[1,0,.1,.1,.1,.1] # Model with no bg 
+    jumpprobability=jumpprobability/sum(jumpprobability)
+
+    # create an RJMCMC structure with all model info
+    acceptfuns=[BAMF.accept_move,BAMF.accept_bg,BAMF.accept_add,BAMF.accept_remove,BAMF.accept_split,BAMF.accept_merge] #array of functions
+    propfuns=[BAMF.propose_move,BAMF.propose_bg,BAMF.propose_add,BAMF.propose_remove,BAMF.propose_split,BAMF.propose_merge] #array of functions
+    myRJMCMC=RJMCMC.RJMCMCStruct(burnin,iterations,njumptypes,jumpprobability,propfuns,acceptfuns)
+end
+#=
+# Create synthetic data
+data=BAMF.ArrayDD(sz)      
+BAMF.genmodel!(datastate,psf,data)
+BAMF.poissrnd!(data.data)
+# imshow(data.data)  #look at data  
+
+=#
```

Comment:
Most of the code above this comment except for the 'using' statements are helper functions for running the tests which are below. I suggest putting these helpers into a separate file to reduce the clutter and make the the test cases easier to find/read/update.